app/models/miq_filter.rb

```diff
@@ -114,8 +117,7 @@ def self.apply_belongsto_filters(inputs, bfilters)
           end
           break if filtered.last == p
         else
-          vcmeta_list = belongsto2object_list(tag)
-          vcmeta_list.pop if vcmeta_list.last.kind_of?(Host)
+          vcmeta_list = vcmeta_list[0..-2] if vcmeta_list.last.kind_of?(Host)
```

Comment:
pop modifies the array in place, so, since we are sharing the Array now instead we just grab a sub-array and leave the original alone.

Reply:
is is possible to make this modification directly in `belongs2object_list`?
better yet, make it before we even hit the database since the relationship records know if it is a host before we even get this far

the use of the host in the storages may be the only part that would require the host?
I think I'm misunderstanding this code

Reply:
No because there are other callers to `belongs2object_list` that need the full list (in particular the other branch of this if statement)